test/use-checkout-pricing.test.js

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect } from 'react';
+import { render, wait, cleanup } from './support/custom-render';
+import { useCheckoutPricing } from '../lib';
+
+const subscriptionPricingReturn = {
+  plan: jest.fn(() => subscriptionPricingReturn),
+  addon: jest.fn(() => subscriptionPricingReturn),
+  tax: jest.fn(() => subscriptionPricingReturn),
+  catch: jest.fn(() => subscriptionPricingReturn),
+  done: jest.fn(),
+};
+
+const mockPricing = {
+  price: {
+    now: {
+      subtotal: '10.00',
+    },
+  },
+};
+
+const checkoutPricingReturn = {
+  subscription: jest.fn(() => checkoutPricingReturn),
+  catch: jest.fn(() => mockPricing),
+  then: jest.fn(() => checkoutPricingReturn),
+  finally: jest.fn(() => checkoutPricingReturn),
+};
+
+const CheckoutPricingFixture = ({ initialInput, nextInput, handleError }) => {
+  const [{ price, pricing, loading }, setPricing] = useCheckoutPricing(initialInput, handleError);
+
+  useEffect(() => {
+    if (nextInput) {
+      setPricing(nextInput);
+    }
+  }, [nextInput]);
+
+  return (
+    <div>
+      <div>{loading ? 'loading' : ''}</div>
+      <div>{price && price.now && price.now.subtotal}</div>
+    </div>
+  );
+};
+
+describe('useCheckoutPricing', function() {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    const { Recurly } = window.recurly;
+
+    window.recurly.Recurly = () => {
+      return {
+        ...new Recurly('Public key'),
+        Pricing: {
+          Subscription: jest.fn(() => subscriptionPricingReturn),
+          Checkout: jest.fn(() => checkoutPricingReturn),
+        },
+      };
+    };
+  });
+
+  it('should call pricing functions with initial inputs', async () => {
+    render(<CheckoutPricingFixture initialInput={{ subscriptions: [{ plan: 'basic', quantity: 2 }] }} />);
+
+    await wait(() => {
```

Comment:
All of the assertions in these tests need to be `awaited` with RTL's `wait` function to comply with a warning that would otherwise emit. Without waiting, the assertions run before state is updated. 

For more information: https://github.com/testing-library/react-testing-library/issues/480